builtin/providers/aws/resource_aws_db_event_subscription.go

```diff
@@ -0,0 +1,270 @@
+		return err
+	}
+	if err := d.Set("event_categories", flattenStringList(sub.EventCategoriesList)); err != nil {
+		return err
+	}
+
+	// Set tags, but theyre not in the response
+
+	return nil
+}
+
+func resourceAwsDbEventSubscriptionRetrieve(
+	d *schema.ResourceData, meta interface{}) (*rds.EventSubscription, error) {
+	rdsconn := meta.(*AWSClient).rdsconn
+	name := d.Get("name").(string)
+	request := &rds.DescribeEventSubscriptionsInput{
+		SubscriptionName: aws.String(name),
+	}
+
+	describeResp, err := rdsconn.DescribeEventSubscriptions(request)
+	if err != nil {
+		if rdserr, ok := err.(awserr.Error); ok && rdserr.Code() == "SubscriptionNotFound" {
+			log.Printf("[WARN] No RDS Event Subscription by name (%s) found", name)
+			d.SetId("")
+			return nil, nil
+		}
+		return nil, fmt.Errorf("Error reading RDS Event Subscription %s: %s", name, err)
+	}
+
+	if len(describeResp.EventSubscriptionsList) != 1 {
+		return nil, fmt.Errorf("Unable to find RDS Event Subscription: %#v", describeResp.EventSubscriptionsList)
+	}
+
+	return describeResp.EventSubscriptionsList[0], nil
+}
+
+func resourceAwsDbEventSubscriptionUpdate(d *schema.ResourceData, meta interface{}) error {
+	rdsconn := meta.(*AWSClient).rdsconn
+
+	// check/update tags, missing from go-aws-sdk?
+	if d.HasChange("enabled") || d.HasChange("event_categories") ||
+		d.HasChange("sns_topic") || d.HasChange("source_type") {
+		eventCategoriesSet := d.Get("event_categories").(*schema.Set)
+		eventCategories := make([]*string, eventCategoriesSet.Len())
+		for i, eventCategory := range eventCategoriesSet.List() {
+			eventCategories[i] = aws.String(eventCategory.(string))
+		}
+
+		// Do we want to wait while it is in "modifying" state?
+
+		_, err := rdsconn.ModifyEventSubscription(&rds.ModifyEventSubscriptionInput{
+			SubscriptionName: aws.String(d.Id()),
+			Enabled:          aws.Bool(d.Get("enabled").(bool)),
+			SnsTopicArn:      aws.String(d.Get("sns_topic").(string)),
+			SourceType:       aws.String(d.Get("source_type").(string)),
+			EventCategories:  eventCategories,
+		})
+
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func resourceAwsDbEventSubscriptionDelete(d *schema.ResourceData, meta interface{}) error {
+	rdsconn := meta.(*AWSClient).rdsconn
+	deleteOpts := rds.DeleteEventSubscriptionInput{
+		SubscriptionName: aws.String(d.Id()),
+	}
+
+	if _, err := rdsconn.DeleteEventSubscription(&deleteOpts); err != nil {
+		rdserr, ok := err.(awserr.Error)
+		if !ok {
+			return err
+		}
+
+		if rdserr.Code() != "DBEventSubscriptionNotFoundFault" {
+			return err
+		}
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"deleting"},
+		Target:     []string{},
+		Refresh:    resourceAwsDbEventSubscriptionRefreshFunc(d, meta),
+		Timeout:    40 * time.Minute,
+		MinTimeout: 10 * time.Second,
+		Delay:      30 * time.Second, // Wait 30 secs before starting
+	}
+	_, err := stateConf.WaitForState()
+	return err
```

Comment:
Can you make this error more specific like you did the one for Create/Describe? e.g. `fmt.Errorf("Deleting RDS Event Subscription %s failed: %s", d.Id(), err)`.

The errors coming from the API in these cases don't tend to be very descriptive.